### ISSUE TITLE
`hab sup start myorigin/myservice` does not start correctly.

### ISSUE THREAD

**Issue by rhass** [habitat-sh/habitat#2110]
`hab sup start core/postgres` seems to load the incorrect spec file by matching whatever is found first in the directory.

Steps to reproduce:
1. `sudo hab sup unload core/postgresql`
2. `sudo hab sup load core/postgresql`
3. `sudo hab sup load core/postgresql2`
4. Note the output of `ls -al /hab/sup/default/specs/`:
```
$ ls -al /hab/sup/default/specs/
total 16
drwxr-xr-x 2 root root 4096 Apr 17 15:20 .
drwxr-xr-x 4 root root 4096 Apr 17 15:13 ..
-rw-r--r-- 1 root root  202 Apr 17 15:20 postgresql2.spec
-rw-r--r-- 1 root root  201 Apr 17 15:19 postgresql.spec
```
5. `sudo hab sup start core/postgresql`

It seems to attempt to load the `postgresql2` spec and since the package `core/postgresql2` doesn't exist, it fails to start. In my opinion there are at least two bugs here.

1. The `hab sup load` command allows us run load against packages which do not exist.
2. The `hab sup start` command pattern matches incorrectly, and should use the exact package specified. 

- [x] The OS (including version) where you are running any of the Habitat commands.
Ubuntu 16.04.2
- [x] Debug/backtrace of the command you are trying to run. You can set the following environment variables before running the `hab` command to generate a trace:

```
hab-sup(MR)[src/manager/mod.rs:725:24]: Unable to start Habitat Supervisor because we weren't able to write or read to a process lock at /hab/sup/default/LOCK, File exists (os error 17)
rhass@rogue-ubuntu:~/src/habitat$ sudo hab sup start core/postgresql
hab-sup(MR): Butterfly Member ID c29632eb2d9f4602ba9f09946088a8bd
hab-sup(SR): Adding core/postgresql/9.6.1/20170215221136
hab-sup(SR): Package core/postgresql2 not found locally, installing from https://willem.habitat.sh/v1/depot
» Installing core/postgresql2
hab-sup(ER)[src/error.rs:337:8]: [404 Not Found]
rhass@rogue-ubuntu:~/src/habitat$ sudo RUST_LOG=debug hab sup start core/postgresql
DEBUG:habitat_common::ui: InputStream(stdin): { is_a_terminal(): true }
DEBUG:habitat_common::ui: OutputStream(stdout): { is_colored(): true, supports_color(): true, is_a_terminal(): true }
DEBUG:habitat_common::ui: OutputStream(stderr): { is_colored(): true, supports_color(): true, is_a_terminal(): true }
DEBUG:habitat_core::os::process::imp: Calling execvp(): ("/hab/pkgs/core/hab-sup/0.20.0/20170407020554/bin/hab-sup") ["start", "core/postgresql"]
DEBUG:habitat_sup::manager: Creating data directory: /hab/sup/default/data
DEBUG:habitat_sup::manager: Creating specs directory: /hab/sup/default/specs
DEBUG:habitat_sup::manager: Cleaning cached health checks
hab-sup(MR): Butterfly Member ID c29632eb2d9f4602ba9f09946088a8bd
DEBUG:habitat_sup::manager::spec_watcher: Service spec for postgresql is new, enqueuing AddService(ServiceSpec { ident: PackageIdent { origin: "core", name: "postgresql", version: None, release: None }, group: "default", depot_url: "https://willem.habitat.sh/v1/depot", topology: Standalone, update_strategy: None, binds: [], config_from: None, desired_state: Up, start_style: Persistent }) event
DEBUG:habitat_sup::manager::spec_watcher: Service spec for postgresql2 is new, enqueuing AddService(ServiceSpec { ident: PackageIdent { origin: "core", name: "postgresql2", version: None, release: None }, group: "default", depot_url: "https://willem.habitat.sh/v1/depot", topology: Standalone, update_strategy: None, binds: [], config_from: None, desired_state: Up, start_style: Persistent }) event
DEBUG:habitat_common::ui: InputStream(stdin): { is_a_terminal(): true }
DEBUG:habitat_sup::manager::spec_watcher: SpecWatcher(/hab/sup/default/specs) thread starting
DEBUG:habitat_common::ui: OutputStream(stdout): { is_colored(): true, supports_color(): true, is_a_terminal(): true }
DEBUG:habitat_common::ui: OutputStream(stderr): { is_colored(): true, supports_color(): true, is_a_terminal(): true }
DEBUG:habitat_sup::manager::service::config: Failed to open user.toml: No such file or directory (os error 2)
DEBUG:habitat_sup::manager::service::config: Failed to open gossip.toml: No such file or directory (os error 2)
DEBUG:habitat_sup::manager::service::config: Looking up environment variable HAB_POSTGRESQL failed: NotPresent
DEBUG:habitat_sup::manager::service::hooks: file_updated not found at /hab/pkgs/core/postgresql/9.6.1/20170215221136/hooks/file_updated, not loading
DEBUG:habitat_sup::manager::service::hooks: suitability not found at /hab/pkgs/core/postgresql/9.6.1/20170215221136/hooks/suitability, not loading
DEBUG:habitat_sup::manager::service::hooks: reload not found at /hab/pkgs/core/postgresql/9.6.1/20170215221136/hooks/reload, not loading
DEBUG:habitat_sup::manager::service::hooks: reconfigure not found at /hab/pkgs/core/postgresql/9.6.1/20170215221136/hooks/reconfigure, not loading
DEBUG:habitat_sup::manager::service::hooks: smoke_test not found at /hab/pkgs/core/postgresql/9.6.1/20170215221136/hooks/smoke_test, not loading
DEBUG:habitat_sup::manager::service::hooks: postgresql.default, Hooks loaded, destination=/hab/svc/postgresql/hooks, templates=/hab/pkgs/core/postgresql/9.6.1/20170215221136/hooks
hab-sup(SR): Adding core/postgresql/9.6.1/20170215221136
DEBUG:habitat_sup::manager::service: Creating svc paths
DEBUG:habitat_sup::manager::service: Creating dir with subdirs: "/hab/svc/postgresql"
DEBUG:habitat_sup::manager::service: Creating dir with subdirs: "/hab/svc/postgresql/config"
DEBUG:habitat_core::util::perm: Attempting to set owner of "/hab/svc/postgresql/config" to "hab"
DEBUG:habitat_sup::manager::service: Creating dir with subdirs: "/hab/svc/postgresql/data"
DEBUG:habitat_core::util::perm: Attempting to set owner of "/hab/svc/postgresql/data" to "hab"
DEBUG:habitat_sup::manager::service: Creating dir with subdirs: "/hab/svc/postgresql/files"
DEBUG:habitat_core::util::perm: Attempting to set owner of "/hab/svc/postgresql/files" to "hab"
DEBUG:habitat_sup::manager::service: Creating dir with subdirs: "/hab/svc/postgresql/hooks"
DEBUG:habitat_sup::manager::service: Creating dir with subdirs: "/hab/svc/postgresql/var"
DEBUG:habitat_core::util::perm: Attempting to set owner of "/hab/svc/postgresql/var" to "hab"
DEBUG:habitat_sup::manager::service: Creating dir with subdirs: "/hab/svc/postgresql/static"
DEBUG:habitat_core::util::perm: Attempting to set owner of "/hab/svc/postgresql/static" to "hab"
DEBUG:habitat_sup::manager::service: Creating dir with subdirs: "/hab/svc/postgresql/logs"
DEBUG:habitat_sup::manager::service: Creating dir with subdirs: "/hab/svc/postgresql/toml"
DEBUG:habitat_common::ui: InputStream(stdin): { is_a_terminal(): true }
DEBUG:habitat_common::ui: OutputStream(stdout): { is_colored(): true, supports_color(): true, is_a_terminal(): true }
DEBUG:habitat_common::ui: OutputStream(stderr): { is_colored(): true, supports_color(): true, is_a_terminal(): true }
hab-sup(SR): Package core/postgresql2 not found locally, installing from https://willem.habitat.sh/v1/depot
DEBUG:habitat_common::command::package::install: install cache_key_path: /hab/cache/keys
DEBUG:habitat_http_client::ssl: Setting CA file for SSL context to: /hab/pkgs/core/cacerts/2016.09.14/20161031044952/ssl/cert.pem
DEBUG:habitat_http_client::api_client: User-Agent: hab-sup/0.20.0/20170407020554 (x86_64-linux; 4.4.0-72-generic)
» Installing core/postgresql2
DEBUG:habitat_http_client::api_client: GET https://willem.habitat.sh/v1/depot/pkgs/core/postgresql2/latest with ApiClient { endpoint: "https://willem.habitat.sh/v1/depot", inner: Client { redirect_policy: FollowAll, read_timeout: Some(Duration { secs: 30, nanos: 0 }), write_timeout: Some(Duration { secs: 30, nanos: 0 }), proxy: None }, proxy: None, target_scheme: "https", user_agent_header: UserAgent("hab-sup/0.20.0/20170407020554 (x86_64-linux; 4.4.0-72-generic)") }
DEBUG:hyper::net: http scheme
DEBUG:hyper::net: https scheme
DEBUG:hyper::http::h1: request line: Get "/v1/depot/pkgs/core/postgresql2/latest" Http11
DEBUG:hyper::http::h1: headers=Headers { Host: willem.habitat.sh
, User-Agent: hab-sup/0.20.0/20170407020554 (x86_64-linux; 4.4.0-72-generic)
, }
DEBUG:hyper::client::response: version=Http11, status=NotFound
DEBUG:hyper::client::response: headers=Headers { Access-Control-Allow-Headers: authorization, range
, Access-Control-Allow-Methods: PUT, DELETE
, Access-Control-Allow-Origin: *
, Server: nginx/1.10.1
, Via: 1.1 varnish
Via: 1.1 varnish
, Fastly-Debug-Digest: 648b5bfebac528318aa21ddbf164a39b6fab1b8dd157e678dfab4fd646dad231
, Content-Length: 0
, Accept-Ranges: bytes
, Date: Mon, 17 Apr 2017 22:21:54 GMT
, Age: 0
, Connection: keep-alive
, X-Served-By: cache-sea1025-SEA, cache-lax8633-LAX
, X-Cache: MISS, MISS
, X-Cache-Hits: 0, 0
, X-Timer: S1492467714.421355,VS0,VE107
, Strict-Transport-Security: max-age= 7776000; includeSubDomains
, X-Frame-Options: SAMEORIGIN
, }
DEBUG:habitat_sup::supervisor: Attempting to clean up pid file /hab/svc/postgresql/PID
DEBUG:habitat_sup::supervisor: Error removing pidfile: No such file or directory (os error 2), continuing
hab-sup(ER)[src/error.rs:337:8]: [404 Not Found]
```

- [x] Current Habitat environment variables where the `hab` command or supervisor is running.  These can be gathered using:
*NONE*

**Comment by reset** [comment]
It looks like these are going to be the steps to solve this issue

* Validate package is installed in `load` command or automatically install it for them (at time of load)
* We shouldn't kill the supervisor if only one of the services fails to start
* Bonus round: Improve validation on package ident in `start`/`load` commands

**Comment by mwrock** [comment]
I just looked through this. The first bug you mention @rhass is crystal clear. The second one I am not seeing. Our package ident location logic is exact and not fuzzy and I'm not seeing `start` commands for postgresql trying to start postgresql2. I loaded both postgresql and postgresql2.

Here is what we are going to do:

1. If `load` results in a 404 attempting to install a package, it will not load the spec.
2. Running `load` will provide feedback to the shell where `load` was called stating either that the package was loaded or that it was not succesful. This second item will be implemented as part of #2121 

@rhass if you think I'm missing something on that second bug, let me know. We did have a release since you last reported this so perhaps that was fixed.

@reset it does not appear that the spuervisor crashes on a failure to start a service

**Comment by rhass** [comment]
Matt and I spoke over Zoom and clarified the second issue which really was a misunderstand on my part on how `hab sup start` and `hab sup load` function when a supervisor is started. Specifically, `hab sup load` will cause anything loaded to attempt to startup, even if `hab sup start` is the command issued.
